pkg/execution/datasource/datasource_http_polling_stream.go

```diff
@@ -110,11 +110,27 @@ type HttpPollingStreamDataSource struct {
 	Log      log.Logger
 	once     sync.Once
 	ch       chan []byte
-	closed   bool
 	Delay    time.Duration
 	client   *http.Client
 	request  *http.Request
 	lastData []byte
+
+	// The mutex guards the fields following it. Use the
+	// accessor methods to read/write them.
+	mu     sync.RWMutex
```

Comment:
Why is this needed? Do we access the field from multiple goroutines?

Reply:
`Resolve()` starts `startPolling()` on a different goroutine. The former reads and writes to the `closed`  field while the latter writes to it